iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/processor/downsampling/changing/ChangingPointSamplingProcessor.java

```diff
@@ -0,0 +1,244 @@
+          parameters.getDoubleOrDefault(
+              PipeProcessorConstant.PROCESSOR_CHANGING_POINT_VALUE_INTERVAL,
+              PipeProcessorConstant.PROCESSOR_CHANGING_POINT_VALUE_INTERVAL_DEFAULT_VALUE);
+      eventTimeMinInterval =
+          parameters.getLongOrDefault(
+              PipeProcessorConstant.PROCESSOR_CHANGING_POINT_EVENT_TIME_MIN_INTERVAL,
+              PipeProcessorConstant.PROCESSOR_CHANGING_POINT_EVENT_TIME_MIN_INTERVAL_DEFAULT_VALUE);
+      eventTimeMaxInterval =
+          parameters.getLongOrDefault(
+              PipeProcessorConstant.PROCESSOR_CHANGING_POINT_EVENT_TIME_MAX_INTERVAL,
+              PipeProcessorConstant.PROCESSOR_CHANGING_POINT_EVENT_TIME_MAX_INTERVAL_DEFAULT_VALUE);
+      arrivalTimeMinInterval =
+          parameters.getLongOrDefault(
+              PipeProcessorConstant.PROCESSOR_CHANGING_POINT_ARRIVAL_TIME_MIN_INTERVAL,
+              PipeProcessorConstant
+                  .PROCESSOR_CHANGING_POINT_ARRIVAL_TIME_MIN_INTERVAL_DEFAULT_VALUE);
+      arrivalTimeMaxInterval =
+          parameters.getLongOrDefault(
+              PipeProcessorConstant.PROCESSOR_CHANGING_POINT_ARRIVAL_TIME_MAX_INTERVAL,
+              PipeProcessorConstant
+                  .PROCESSOR_CHANGING_POINT_ARRIVAL_TIME_MAX_INTERVAL_DEFAULT_VALUE);
+    } else {
+      isFilteredByArrivalTime = false;
+      compressionDeviation =
+          parameters.getDoubleOrDefault(
+              PipeProcessorConstant.PROCESSOR_CHANGING_VALUE_COMPRESSION_DEVIATION,
+              PipeProcessorConstant.PROCESSOR_CHANGING_POINT_VALUE_INTERVAL_DEFAULT_VALUE);
+      eventTimeMinInterval =
+          parameters.getLongOrDefault(
+              PipeProcessorConstant.PROCESSOR_CHANGING_VALUE_MIN_TIME_INTERVAL_KEY,
+              PipeProcessorConstant.PROCESSOR_CHANGING_VALUE_MIN_TIME_INTERVAL_DEFAULT_VALUE);
+      eventTimeMaxInterval =
+          parameters.getLongOrDefault(
+              PipeProcessorConstant.PROCESSOR_CHANGING_VALUE_MAX_TIME_INTERVAL_KEY,
+              PipeProcessorConstant.PROCESSOR_CHANGING_VALUE_MAX_TIME_INTERVAL_DEFAULT_VALUE);
+      // will not be used
+      arrivalTimeMinInterval = 0;
+      arrivalTimeMaxInterval = Long.MAX_VALUE;
+    }
+
+    validatorTimeInterval(validator);
+    validator.validate(
+        compressionDeviation -> (Double) compressionDeviation >= 0,
+        String.format(
+            "%s must be >= 0, but got %s",
+            PipeProcessorConstant.PROCESSOR_CHANGING_VALUE_COMPRESSION_DEVIATION,
+            compressionDeviation),
+        compressionDeviation);
+  }
+
+  @Override
+  public void customize(
+      PipeParameters parameters, PipeProcessorRuntimeConfiguration configuration) {
+    super.customize(parameters, configuration);
+
+    final boolean isChangingPointProcessor =
+        BuiltinPipePlugin.CHANGING_POINT_SAMPLING_PROCESSOR
+            .getPipePluginName()
+            .equals(parameters.getString("processor"));
+
+    if (isChangingPointProcessor) {
```

Comment:
Do we need to separate the logger? It seems that the info log is mostly for ourselves....